180-API-MultipleItems

Patch to extend API to get multiple items
Available routes:
  POST /api/v1/biblios/items

  takes array of biblionumbers as BODY

diff --git a/Koha/REST/V1/MultipleItems.pm b/Koha/REST/V1/MultipleItems.pm
new file mode 100644
index 0000000..6ca1d9c
--- /dev/null
+++ b/Koha/REST/V1/MultipleItems.pm
@@ -0,0 +1,47 @@
+package Koha::REST::V1::MultipleItems;
+
+# This file is part of Koha.
+#
+# Koha is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 3 of the License, or (at your option) any later
+# version.
+#
+# Koha is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with Koha; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+use Modern::Perl;
+
+use Mojo::Base 'Mojolicious::Controller';
+
+use C4::Auth qw( haspermission );
+use Koha::Biblios;
+
+use Data::Dumper;
+
+sub get {
+    my ($c, $args, $cb) = @_;
+
+    my $json = $c->req->json;
+    warn Dumper($json);
+    unless ($json) {
+        return $c->$cb({error => "Missing JSON body"}, 400);
+    }
+    my @items = ();
+    for my $biblionumber (@{$json}) {
+        my $biblio = Koha::Biblios->find($biblionumber);
+        if ($biblio) {
+            warn Dumper($biblio->items->unblessed);
+            push @items, $biblio->items->unblessed;
+        }
+
+    }
+    return $c->$cb({ items => \@items }, 200);
+}
+
+1;
diff --git a/api/v1/swagger/paths.json b/api/v1/swagger/paths.json
index e81c59f..27de867 100644
--- a/api/v1/swagger/paths.json
+++ b/api/v1/swagger/paths.json
@@ -1,4 +1,7 @@
 {
+  "/biblios/items": {
+    "$ref": "paths/multipleitems.json#/~1biblios~1items"
+  },
   "/messaging/patrons/{borrowernumber}/accountdetails": {
     "$ref": "paths/messaging.json#/~1messaging~1patrons~1{borrowernumber}~1accountdetails"
   },
diff --git a/api/v1/swagger/paths/multipleitems.json b/api/v1/swagger/paths/multipleitems.json
new file mode 100644
index 0000000..b931be0
--- /dev/null
+++ b/api/v1/swagger/paths/multipleitems.json
@@ -0,0 +1,26 @@
+{
+  "/biblios/items": {
+    "post": {
+      "operationId": "getMultipleItems",
+      "tags": ["items"],
+      "produces": [ "application/json" ],
+      "responses": {
+        "200": {
+          "description": "An array of items"
+        },
+        "400": {
+          "description": "Error in request",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        },
+        "403": {
+          "description": "Access forbidden",
+          "schema": {
+            "$ref": "../definitions.json#/error"
+          }
+        }
+      }
+    }
+  }
+}
